platforms/core-configuration/model-core/src/main/java/org/gradle/internal/snapshot/impl/IsolatedArrayOfPrimitive.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright 2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.snapshot.impl;
+
+import org.gradle.api.NonNullApi;
+import org.gradle.internal.hash.Hashable;
+import org.gradle.internal.hash.Hasher;
+import org.gradle.internal.isolation.Isolatable;
+import org.gradle.internal.snapshot.ValueSnapshot;
+import org.gradle.internal.snapshot.ValueSnapshottingException;
+
+import javax.annotation.Nullable;
+
+import static org.gradle.internal.snapshot.impl.AbstractValueProcessor.javaSerialized;
+import static org.gradle.internal.snapshot.impl.JavaSerializedValueSnapshot.javaDeserialized;
+
+@NonNullApi
+public class IsolatedArrayOfPrimitive implements Isolatable<Object>, Hashable {
+    private final PrimitiveType primitiveType;
+    private final Object array;
+
+    public IsolatedArrayOfPrimitive(Object array) {
+        this(PrimitiveType.of(array.getClass()), array);
+    }
+
+    private IsolatedArrayOfPrimitive(PrimitiveType primitiveType, Object array) {
+        this.primitiveType = primitiveType;
+        this.array = array;
+    }
+
+    public static IsolatedArrayOfPrimitive fromByteArray(byte primitiveTypeCode, byte[] bytes) {
+        PrimitiveType primitiveType = PrimitiveType.fromOrdinal(primitiveTypeCode);
+        return new IsolatedArrayOfPrimitive(
+            primitiveType,
+            fromByteArray(bytes, primitiveType)
+        );
+    }
+
+    @Override
+    public void appendToHasher(Hasher hasher) {
+        throw unsupportedOperation();
+    }
+
+    @Override
+    public ValueSnapshot asSnapshot() {
+        throw unsupportedOperation();
+    }
+
+    private UnsupportedOperationException unsupportedOperation() {
+        return new UnsupportedOperationException("This operation is not supported on primitive arrays of type: " + primitiveType.arrayType);
+    }
+
+    @Override
+    public Object isolate() {
+        switch (primitiveType) {
+            case B:
+                return ((byte[]) array).clone();
```

Comment:
:thinking: I wonder if this and other two switches can be replaced with a couple of virtual methods on the PrimitiveType enum.

Reply:
Done in 48d173769b0fcd2ea3b7aae7e67d5673bee89640